src/renderer/components/FileUploadStages/ProcessFiles/Stage3.tsx

```diff
@@ -97,22 +114,155 @@ const CheckProgress = (props: CheckProgressProps) => {
  */
 const Stage3 = (props: Stage3Props) => {
     const classes = useStyles();
-    const {files} = props;
+    const { files } = props;
 
     // React state variable and setter that keeps track of the current file index
     const [currentFileIndex, setCurrentFileIndex] = React.useState<number>(0);
+    const [responseObjects, setResponseObjects] = React.useState<Array<JhoveValidationResponse>>([]);
+    const [processFinished, setFinishState] = React.useState<boolean>(false);
+    const [showResult, setShowResult] = React.useState<boolean>(false);
+
+
+    /**
+     * Validates the next file in the array of files using JHOVE-REST. This function makes use of the
+     * currentFileIndex, responseObjects and files state objects.
+     */
+    const validateNextFile = () => {
+        if (currentFileIndex < files.length && !processFinished) {
+            let formData = new FormData();
+            formData.append('module', 'TIFF-opf');
+            formData.append('file', files[currentFileIndex]);
+            axios.post(JHOVE_API("api/jhove/validate"), formData)
+                .then((res: AxiosResponse) => {
+                    let data: JhoveValidationResponse = res.data;
+                    setResponseObjects([...responseObjects, data]);
+                    setCurrentFileIndex(currentFileIndex + 1);
+                });
+        }
+        else {
+            setFinishState(true);
+        }
+    }
+
+    /**
+     * Hook that runs on update of the props currentFileIndex and processFinished. 
+     * Validates the files on update. If the process has finished, the steps and files redux state
+     * properties are reset.
+     */
+    React.useEffect(() => {
+        validateNextFile();
+
+        // Function called upon unmounting.
+        // Function also gets called in between render cycles.
+        // Only reset progress if rendering is done == the checking process has finished
+        return () => {
+            if (processFinished) {
+                props.resetStep();
+                props.clearFiles();
+            }
+        }
 
-    // TODO -> Add hooks to connect to JHOVE backend here 
+    }, [currentFileIndex, processFinished]);
+
+    /**
+     * Filter the messages of the JHOVE validation response.
+     * @param response the JHOVE validation response to filter the messages from
+     * @param type the type of message to filter on.
+     */
+    const filterMessages = (response: JhoveValidationResponse, type: "info" | "error" | "warning") => {
+        switch (type) {
+            case "info":
```

Comment:
I think it's a good idea to declare some constants for info, error and warning. You're using them all on line 172 and then again for every case.